### DISCUSSION TITLE
Proper Updates & Keeping Config

### DISCUSSION THREAD

**Discussion by Oskar-Idland** [prasanthrangan/hyprdots#1215] [Q&A] [answered]
I read the wiki and just pulled the latest commit. I have edited the `restore_cfg.lst` to say NO (N) in the first column to not update things like my kitty config. 
1. When I pull, I get a git conflict. How do you deal with that so it does not come in the way of future updates or edits? See`restore_cfg.lst` below:
```sh
───────┬────────────────────────────────────────────────────────────────────────────────────
       │ File: restore_cfg.lst
───────┼────────────────────────────────────────────────────────────────────────────────────
   1   │ N|Y|${HOME}|.zshrc .p10k.zsh|zsh oh-my-zsh-git zsh-theme-powerlevel10k pokemon-colo
       │ rscripts-git
   2   │ N|Y|${HOME}/.config/Code/User|settings.json|visual-studio-code-bin
   3   │ N|Y|${HOME}/.config|dunst|dunst
   4   │ N|Y|${HOME}/.config|fish|fish
   5   │ Y|Y|${HOME}/.config/hypr|animations.conf hyprland.conf theme.ctl windowrules.conf s
       │ cripts themes wallbash|hyprland
   6   │ N|Y|${HOME}/.config/hypr|keybindings.conf monitors.conf userprefs.conf|hyprland
   7   │ N|Y|${HOME}/.config/hypr|nvidia.conf|hyprland nvidia-utils
   8   │ N|Y|${HOME}/.config|kitty|kitty
   9   │ Y|Y|${HOME}/.config|Kvantum|kvantum
  10   │ Y|Y|${HOME}/.config|lsd|lsd
  11   │ Y|Y|${HOME}/.config|MangoHud|mangohud
  12   │ N|Y|${HOME}/.config|neofetch|neofetch
  13   │ Y|Y|${HOME}/.config|qt5ct|qt5ct
  14   │ Y|Y|${HOME}/.config|rofi|rofi
  15   │ Y|Y|${HOME}/.config|swaylock|swaylock-effects
  16   │ Y|Y|${HOME}/.config|swww|swww
  17   │ Y|Y|${HOME}/.config|waybar|waybar
  18   │ Y|Y|${HOME}/.config|wlogout|wlogout
  19   │ Y|Y|${HOME}/.config|xsettingsd nwg-look gtk-3.0|nwg-look
  20   │ Y|Y|${HOME}/.icons|default|nwg-look
  21   │ Y|Y|${HOME}|.gtkrc-2.0|nwg-look
  22   │ Y|Y|${HOME}/.config|dolphinrc kdeglobals|dolphin
  23   │ Y|Y|${HOME}/.local/share|dolphin|dolphin
  24   │ Y|Y|${HOME}/.local/share/kxmlgui5|dolphin|dolphin
───────┴────────────────────────────────────────────────────────────────────────────────────
```
2. When using `./install.sh -r` it still overwrites my configs, removes custom images added to themes and removes themes added through theme patcher. What do I have to do to keep all of this? I noticed my `cfg_backups` folder had many folders in it, but no actual files as seen below:
```sh
cfg_backups/
├── 231229_22h54m06s
├── 240114_13h16m28s
├── 240326_21h25m55s
├── 240326_22h00m57s
├── 240326_23h21m02s
└── 240329_02h29m35s

7 directories, 0 files
```

**Comment by kRHYME7** [comment, marked as answer]
It means "restore" from the Git repository.

Try to use ``` Hyprdots restore defaults ``` it will ask yo 3 things. what to preserve, To sync, and to overwrite.

**Reply by Oskar-Idland** [reply]
Thank you! Had to upgrade as well. I noticed when using the command that when doing "mark all" in the fzf menu, it would also select the empty line below the last option. See attachment below. Might not cause problems. Just a heads up! Thank our for quick responses and great help. You carry this project quite a lot! 
![image](https://github.com/prasanthrangan/hyprdots/assets/69601310/db01f1ae-7712-4d98-910d-25eaa062ec5a)

**Reply by kRHYME7** [reply]
> empty line below the last option.

That's fine it is an intended option to Continue. I'll do a second refactor for this which is a simple fzf menu with the 3 Selection


>  You carry this project quite a lot! 

Hoping many users would help other too! A stone across the water to create many ripples... 😄

**Reply by Oskar-Idland** [reply]
Would love to!
